R/dataset-stream.R

```diff
@@ -0,0 +1,59 @@
+#' Tools for working with streaming data
+#'
+#' Crunch allows you to stream data to a dataset. Streaming data is useful for
+#' datasets which have frequent updates (see 
+#' [the Crunch documentation](http://docs.crunch.io/#streaming-rows) for more)
+#' information. `pendingMessages()` retrieves the number of pending messages and `appendStreamedRows()` appends all pending streamed rows to the dataset.
+#'
+#' @param x a CrunchDataset
+#' @param ... Additional batch attributes to set when appending pending stream 
+#' data. Options include
+#' `name` and `description`.
+#' @return number of pending messages for `pendingMessages()` and the dataset 
+#' with appended messages `appendStreamedRows()`
+#' @name streaming
+#' @aliases streaming pendingMessages appendStreamedRows
+#' @examples
+#' \dontrun{
+#' # need examples!
+#' }
+NULL
+
+
+#' @rdname streaming
+#' @export
+setMethod("pendingMessages", "CrunchDataset", function (x) {
```

Comment:
I'm not sure if "pendingMessages" is intuitive to a user. I feel like it should have "stream" in the function name somewhere.

Also, matter of preference, and I think both patterns exist in the package, but in cases like this where you're adding a "method" for only one object type, sometimes I'll just make it a function and have the first line be `stopifnot(is.dataset(x))`. That may be out of laziness on my part, but it may be marginally more readable than having to parse the `setMethod` syntax. Definitely your call but I wanted to mention the alternative.